core/src/main/java/io/jenkins/pluginhealth/scoring/probes/GitHubOpenIssuesProbe.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.kohsuke.github.GHRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(AbstractOpenIssuesProbe.ORDER)
+class GitHubOpenIssuesProbe extends AbstractOpenIssuesProbe {
+    public static final String KEY = "github-open-issues";
+    private static final Logger LOGGER = LoggerFactory.getLogger(GitHubOpenIssuesProbe.class);
+
+    /**
+     * Get total number of open GitHub issues in a plugin.
+     *
+     * @param context {@link ProbeContext}
+     * @return Optional an Integer value will give total count of open issues.
+     */
+    @Override
+    Optional<Integer> getCountOfOpenIssues(ProbeContext context) {
+        /* Stores the GitHub URL to view all existing issues in the plugin. Ex: https://github.com/jenkinsci/cloudevents-plugin/issues */
+        String issueTrackerViewUrl = context.getIssueTrackerUrlsByNames().get("github");
+
+        if (issueTrackerViewUrl == null) {
+            LOGGER.info("The plugin does not use GitHub issues to track issues.");
```

Comment:
Hi @aneveux 

About adding the names of the plugin, in the logger message, Adrien told me that I shouldn't be passing a parameter to the method just to add it in the logger message. (I had made the mistake in this probe itself and I am not proud of it)

`ProbeContext` does not contain the plugin details like names. I would need to pass the `Plugin` parameter here which is required only in the logs. I do not think this is the best approach. 

After the logs are printed and when the `ProbeResult` is returned as the final result, I have added the plugin name in both the cases. Failure as well as the success message. 

So when the user reads the logs and sees the final `ProbeResult` message I am sure it would make sense to them. 

I hope I understood your concern correctly. If I am on the wrong track here do let me know :)

Reply:
Is there a log message with the plugin name at the beginning of the process?

Something like "Running probes for plugin XXX" and then you have all the logs of the probes execution for that XXX plugin?

I do understand the concern of having the Plugin as parameter just for logging, indeed not a good thing to do.

Anyway, that's probably nitpicking then, so let's not focus too much on that.

Reply:
> Is there a log message with the plugin name at the beginning of the process?

Not really. In the project I have not seen info log at the beginning of the probe. I am not sure if I should add this.

Reply:
Let's not take care of that in that Pull Request then, could be something to have a look at later on while harmonizing the log messages.